src/main/scala/mesosphere/marathon/storage/migration/Migration.scala

```diff
@@ -51,13 +51,25 @@ class Migration(
 
   private[migration] lazy val legacyStoreFuture: Future[Option[PersistentStore]] = legacyConfig.map { config =>
     val store = config.store
+    if (!store.isOpen) {
+      store.markOpen()
+    }
     store match {
       case s: PersistentStoreManagement with PrePostDriverCallback =>
-        s.preDriverStarts.flatMap(_ => s.initialize()).map(_ => Some(store))
+        s.preDriverStarts
+          .flatMap(_ => s.sync())
+          .map(_ => s.initialize())
```

Comment:
What do you think about using `andThen` for side effects here?

Reply:
I don't think it is applicable here.

```
@ Future { 5 } andThen { case _ => throw new Exception("runtime exception") } 
java.lang.Exception: runtime exception
	at ammonite.$sess.cmd8$$anonfun$1.applyOrElse(cmd8.sc:1)
	at ammonite.$sess.cmd8$$anonfun$1.applyOrElse(cmd8.sc:1)
	at scala.concurrent.Future.$anonfun$andThen$1(Future.scala:531)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:29)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:29)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:140)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.pollAndExecAll(ForkJoinPool.java:1021)
	at java.util.concurrent.ForkJoinPool$WorkQueue.execLocalTasks(ForkJoinPool.java:1046)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1058)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
res8: Future[Int] = Future(Success(5))
```

From its docstring:

```
   *  Note that if one of the chained `andThen` callbacks throws
   *  an exception, that exception is not propagated to the subsequent `andThen`
   *  callbacks. Instead, the subsequent `andThen` callbacks are given the original
   *  value of this future.
```

Even worse:

```
@ Future { 5 } andThen { case _ => throw new Exception("runtime exception") } andThen { case Success(n) => println(n); case Failure(f) => println(f) } 
java.lang.Exception: runtime exception
	at ammonite.$sess.cmd12$$anonfun$1.applyOrElse(cmd12.sc:1)
	at ammonite.$sess.cmd12$$anonfun$1.applyOrElse(cmd12.sc:1)
	at scala.concurrent.Future.$anonfun$andThen$1(Future.scala:531)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:29)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:29)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:140)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.pollAndExecAll(ForkJoinPool.java:1021)
	at java.util.concurrent.ForkJoinPool$WorkQueue.execLocalTasks(ForkJoinPool.java:1046)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1058)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
5
res12: Future[Int] = Future(Success(5))
```

It prints `5` and returns `Success(5)` anyway